lib/AppConfig.php

```diff
@@ -72,4 +74,18 @@ public function getUserValue($userId, $key) {
 	public function setUserValue($userId, $key, $value) {
 		return $this->config->setAppValue($userId, $this->appName, $key, $value);
 	}
+
+	/**
+	 * Check if app can have enterprise features enabled
+	 *
+	 * @return bool
+	 */
+	public function enterpriseFeaturesEnabled() {
+		if ($this->config->getAppValue('enterprise_key', 'license-key', null) === null &&
```

Comment:
not enough, you need to also check whether the "enterprise_key" app is enabled which actually verifies the license key. if license key is empty or invalid, the app disables itself

Reply:
I think this is only about soft limit - with invalid key supplied, proper notification will be shown

Reply:
for all other apps the apps will auto-disable if the key is invalid, so here I think the feature should be disabled as well.

@pmaier1 or was this on purpose ?

Reply:
@PVince81 the feature will be disabled without "any" key - valid or invalid. If someone puts invalid, ownCloud will scream that invalid key is supplied

Reply:
an invalid key will auto-disable the enterprise_key and also auto-disable any enterprise apps, no screaming.

Reply:
@PVince81 I can do whatever action is needed, can you point me? I personally would prefer just a soft limit.

Reply:
I can't find anything in our discussions about the desired approach.

I'd prefer to do the check like we do it everywhere else, check that the app is enabled.

Reply:
Agree with @PVince81. The feature (in settings as well as in the sharing dialog and backend) should only be available when `enterprise_key` is enabled. Using the same logic as other EE apps is absolutely fine. We accept that it's easy to circumvent the restriction if you know what you're doing.

Reply:
solved now